### PR TITLE
fix(table): custom background hover colors

### LINKED ISSUE #1454: [bug] `<rh-table>` --rh-table-background-color not working
### Describe the bug

Setting `--rh-table-background-color` has no effect. the property is actively set in the lighdom stylesheet. if we want it to be customizable, it should be documented and use fallbacks

as well, it does not appear in the cem docs

### Which Component?

rh-table

### Steps to reproduce

1. add an rh table with content 
2. add this css
  ```css
  :root {
  --rh-table-background-color: red;
  }
  ```
3. hover a row

### Expected behaviour

bg is red

### Screenshots

_No response_

### Element HTML

_No response_

### Operating System (OS)

Linux

### Browser

_No response_

### More browser information

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1.  Correctly implemented fallbacks for public api css props, added to cem docs, renamed and deprecated old var name.

Closes #1454 

## TODO

- [x] Figure out how to properly document the deprecate on the @cssprop for cem docs?
- [x] Add changeset

## Testing Instructions

1.

### CODE REVIEW

File: docs/release-notes.md
Comment (bennypowers) [inline docs/release-notes.md:1]: looks like maybe you branched off of staging/chansey, can you rebase to main?

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: docs/release-notes.md
Comment (zeroedin) [reply docs/release-notes.md:1]: Ok I think I resolved it, had to force push though FYI.

**Review by bennypowers — COMMENTED** [review]
Needs a changeset then we're g2g

**Comment by zeroedin** [comment]
> Needs a changeset then we're g2g

I couldn't figure out a better way to document the deprecation then just adding the deprecation note to the description on the token in the jsdoc.  Is this something we could improve in the future customizing the format?

**Review by bennypowers — APPROVED** [review]

**Pull request merged**